src/test/app/SetRegularKey_test.cpp

```diff
@@ -30,33 +30,120 @@ class SetRegularKey_test : public beast::unit_test::suite
 
-        env(noop(alice));
         env(noop(alice), sig(bob));
         env(noop(alice), sig(alice));
 
-        // Regular key only
+        testcase("Disable master key");
         env(fset(alice, asfDisableMaster), sig(alice));
-        env(noop(alice));
         env(noop(alice), sig(bob));
         env(noop(alice), sig(alice), ter(tefMASTER_DISABLED));
+
+        testcase("Re-enable master key");
         env(fclear(alice, asfDisableMaster), sig(alice), ter(tefMASTER_DISABLED));
+
+        env(fclear(alice, asfDisableMaster), sig(bob));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice));
+
+        testcase("Revoke regular key");
+        env(regkey(alice, disabled));
+        env(noop(alice), sig(bob), ter(tefBAD_AUTH_MASTER));
+        env(noop(alice), sig(alice));
+    }
+
+    void testDisableMasterKeyAfterFix()
+    {
+        using namespace test::jtx;
+
+        testcase("Set regular key");
+        Env env{*this, supported_amendments() | fixDisabledRegularKey};
+        Account const alice("alice");
+        Account const bob("bob");
+        env.fund(XRP(10000), alice, bob);
+
+        env(regkey(alice, bob));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice));
+
+        testcase("Disable master key");
+        env(fset(alice, asfDisableMaster), sig(alice));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice), ter(tefMASTER_DISABLED));
+
+        testcase("Re-enable master key");
+        env(
+            fclear(alice, asfDisableMaster),
+            sig(alice),
+            ter(tefMASTER_DISABLED)
+        );
+
         env(fclear(alice, asfDisableMaster), sig(bob));
+        env(noop(alice), sig(bob));
+        env(noop(alice), sig(alice));
+
+        testcase("Revoke regular key");
+        env(regkey(alice, disabled));
+        env(noop(alice), sig(bob), ter(tefBAD_AUTH));
         env(noop(alice), sig(alice));
     }
 
+    void testDisabledRegularKey()
+    {
+        using namespace test::jtx;
+
+        // See https://ripplelabs.atlassian.net/browse/RIPD-1721.
+        testcase("Set regular key to master key (before fixDisabledRegularKey)");
+        Env env {*this, supported_amendments() - fixDisabledRegularKey};
+        Account const alice("alice");
+        env.fund(XRP(10000), alice);
+
+        // Must be possible unless amendment `fixDisabledRegularKey` enabled.
+        // log << "set regular key to master key" << std::endl;
+        env(regkey(alice, alice), sig(alice));
+        // log << "disable master key" << std::endl;
+        env(fset(alice, asfDisableMaster), sig(alice));
+
+        // No way to sign...
+        // log << "implicitly sign transaction with master key" << std::endl;
+        env(noop(alice), ter(tefMASTER_DISABLED));
+        env(noop(alice), sig(alice), ter(tefMASTER_DISABLED));
+
+        // ... until now.
+        // log << "pass amendment: fixDisabledRegularKey" << std::endl;
```

Comment:
Go ahead and remove these commented out `log` lines. Chances are they won't be needed again.